packages/notifi-react/lib/context/NotifiTargetContext.tsx

```diff
@@ -310,16 +316,74 @@ export const NotifiTargetContextProvider: FC<PropsWithChildren> = ({
       .filter((item): item is Target => !!item);
   }, [targetInfoPrompts, targetData]);
 
+  const EMAIL_REGEX = /^[a-zA-Z0-9._:$!%-+]+@[a-zA-Z0-9.-]+.[a-zA-Z]$/;
+  const TELEGRAM_REGEX =
+    /^@?(?=\w{5,32}\b)[a-zA-Z0-9]+(?:[a-zA-Z0-9_ ]+[a-zA-Z0-9])*$/;
+
+  const validateEmail = (
+    email: string,
+  ): { isValid: boolean; error?: string } => {
+    if (email === '') {
+      return { isValid: true };
+    }
+    const isValid = EMAIL_REGEX.test(email);
+    return {
+      isValid,
+      error: isValid ? undefined : 'Invalid email address',
+    };
+  };
+
+  const validateTelegram = (
+    telegramId: string,
+  ): { isValid: boolean; error?: string } => {
+    if (telegramId === '') {
+      return { isValid: true };
+    }
+    const isValid = telegramId.length > 1 && TELEGRAM_REGEX.test(telegramId);
+    return {
+      isValid,
+      error: isValid ? undefined : 'Invalid telegram ID',
+    };
+  };
+
   const updateTargetInputs = useCallback(
     <T extends 'form' | 'toggle'>(
       target: T extends 'form' ? FormTarget : ToggleTarget,
-      value: T extends 'form' ? { value: string; error?: string } : boolean,
+      value: T extends 'form' ? TargetInputFromValue : boolean,
     ) => {
       if (target in targetInputs) {
-        setTargetInputs((prev) => ({
-          ...prev,
-          [target]: value,
-        }));
+        setTargetInputs((prev) => {
+          if (typeof value === 'object' && 'value' in value) {
+            if (target === 'email') {
+              const { isValid, error } = validateEmail(value.value);
+              return {
+                ...prev,
+                [target]: { value: value.value, error, isValid },
+              };
+            } else if (target === 'telegram') {
+              const { isValid, error } = validateTelegram(value.value);
+              return {
+                ...prev,
+                [target]: { value: value.value, error, isValid },
+              };
+            } else {
+              // For other form inputs (e.g., phoneNumber)
+              return {
+                ...prev,
+                [target]: { value: value.value, error: value.error },
+              };
+            }
+          } else if (typeof value === 'boolean') {
+            // For toggle inputs
+            return {
+              ...prev,
+              [target]: value,
+            };
+          }
+          // If we reach here, it means we got an unexpected input
+          console.error(`Unexpected input for target ${target}:`, value);
+          return prev; // Return previous state unchanged
+        });
```

Comment:
The same comment as above.  I would not put this logic in Context. Context should be subjective. 
Defining the valid or invalid should be outside of context.
See the example [here](https://github.com/notifi-network/notifi-sdk-ts/blob/79b728739e350e118da59a3305f05e9cef1dc419/packages/notifi-react/lib/components/TargetInputs.tsx#L63)